### ISSUE TITLE
Fail to Find Empty Droplets

### ISSUE THREAD

**Issue by ZTYyy** [broadinstitute/CellBender#144]
It seems that CellBender can not find the correct empty droplets on my data. The UMI curve is like this.

![1660041761335](https://user-images.githubusercontent.com/12484357/183629199-3cd90aca-ff0e-4b78-a500-cbe8d722cf93.png)

and when I run like this 

--input ./raw --output ./cellbender/X-N2.cellbender.h5 --expected-cells 1420 --total-droplets-included 146539 --cuda

Cell output

cellbender:remove-background: Prior on counts in empty droplets is 98
cellbender:remove-background: Prior on counts for cells is 10965
cellbender:remove-background: Excluding barcodes with counts below 49
cellbender:remove-background: Using 1420 probable cell barcodes, plus an additional 48082 barcodes, and 0 empty droplets.
cellbender:remove-background: Largest surely-empty droplet has 49 UMI counts.

Then I read some issues on Github reporting similar problems, and change my command like this

--input ./raw --output ./cellbender_new/X-N2.cellbender_new.h5 --expected-cells 1420 --total-droplets-included 30000 --low-count-threshold 100 --cuda

Cell output

cellbender:remove-background: Prior on counts in empty droplets is 108
cellbender:remove-background: Prior on counts for cells is 10965
cellbender:remove-background: Excluding barcodes with counts below 100
cellbender:remove-background: Using 1420 probable cell barcodes, plus an additional 25522 barcodes, and 0 empty droplets.
cellbender:remove-background: Largest surely-empty droplet has 100 UMI counts.
cellbender:remove-background: Warning: few empty droplets identified. Low UMI cutoff may be too high. Check the UMI decay curve, and decrease the --low-count-threshold parameter if necessary.

I also try on other data, and they all showed that CellBender took the tails as empty droplets, not the right part.

Besides, my matrix is generated from STARsolo, and I am sure I input the raw matrix not the filtered.

**Comment by sjfleming** [comment]
Looks like a good dataset!  If I were you, I would try

```console
cellbender remove-background --input ./raw --output ./cellbender_new/X-N2.cellbender_new.h5 --expected-cells 1420 --total-droplets-included 20000 --low-count-threshold 30 --cuda
```

and see what that gives you.  My thinking is:
- 20000 droplets gets you into the "surely empty" region of the UMI curve, i.e. every droplet past 20k is "surely empty"
- 30 is probably a safe bet for `--low-count-threshold`, since everything with UMI count < 30 is beyond the "empty droplets" and out into the part of the UMI curve that probably represents uncorrected barcode sequencing errors / ambient capture oligos from uncaptured barcodes
  - Although you can probably just use the default of 15 as well, so you might not need to specify `--low-count-threshold`

**Comment by sjfleming** [comment]
Automated heuristics in v0.3.0 should work well for samples like this in future.

Closed by #238
